sippy-ng/src/components/TimelineChart.js

```diff
@@ -0,0 +1,130 @@
+import * as d3scale from 'd3-scale'
+import PropTypes from 'prop-types'
+import React, { useEffect, useRef } from 'react'
+import TimelinesChart from 'timelines-chart'
+
+// TimelineChart is a React component to wrap the plain TimelinesChart we used
+// in origin previously.
+export default function TimelineChart({ eventIntervals, data }) {
+  const ref = useRef(null)
+
+  useEffect(() => {
+    let chart = null
+    const node = ref.current
+    if (node) {
+      //const el = document.querySelector('#chart')
+      let ordinalScale = d3scale
```

Comment:
I think this could be a `const` outside of useEffect, it won't change if data changes.